### PR TITLE
refactor etherscan.io links

### PR DESCRIPTION
**Pull request by derhuerst**
#2855 

Please check that I haven't missed anything.

### CODE REVIEW

File: js/src/3rdparty/etherscan/addressLink.js [hunk 1]
```diff
@@ -14,8 +14,6 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-// links to chain explorers
-export const BASE_LINK_ACCOUNT_MORDEN = 'https://testnet.etherscan.io/address/';
-export const BASE_LINK_ACCOUNT_HOMESTEAD = 'https://etherscan.io/address/';
-export const BASE_LINK_TX_MORDEN = 'https://testnet.etherscan.io/tx/';
-export const BASE_LINK_TX_HOMESTEAD = 'https://etherscan.io/tx/';
+export default (address, isTestnet = false) => {
```
Comment (jacogr) [inline js/src/3rdparty/etherscan/addressLink.js]: Put txLint & addressLink functions into a single file, links.js (aligns with the rest and doesn't end up with 90% of the files have more boilerplate than actual code)

File: js/src/3rdparty/etherscan/call.js [hunk 2]
```diff
@@ -23,19 +25,14 @@ const options = {
...
-  let params = '';
-
-  if (_params) {
-    Object.keys(_params).map((param) => {
-      const value = _params[param];
 
-      params = `${params}&${param}=${value}`;
-    });
-  }
+  const query = stringify(Object.assign({
```
Comment (jacogr) [inline js/src/3rdparty/etherscan/call.js]: Definitely cleaner, always considered this the better option, just didn't want to put in the qs module for 4 single lines.

**Comment by jacogr** [comment]
Couple of minor comments, as long as the find-everything-containing-etherscan turns up empty, should have all the case.

File: js/src/3rdparty/etherscan/call.js (hunk 2 above)
Comment (derhuerst) [inline js/src/3rdparty/etherscan/call.js]: Yeah, I also thought about this and in doubt, I'm always for the more lightweight/simpler solution. But in this case, things will actually break with perfectly valid api calls.

File: js/src/3rdparty/etherscan/addressLink.js (hunk 1 above)
Comment (derhuerst) [inline js/src/3rdparty/etherscan/addressLink.js]: Will do.

File: js/src/views/Signer/components/TxHashLink/TxHashLink.js [hunk 3]
```diff
@@ -27,37 +27,17 @@ export default class TxHashLink extends Component {
...
-
   render () {
-    const { children, txHash, className } = this.props;
-    const { link } = this.state;
+    const { children, txHash, className, chain } = this.props;
 
     return (
       <a
-        href={ link }
+        href={ txLink(txHash, chain === 'morden' || chain === 'testnet') }
```
Comment (jacogr) [inline js/src/views/Signer/components/TxHashLink/TxHashLink.js]: Any reason why are not just pulling in isTest here?

Comment (derhuerst) [inline js/src/views/Signer/components/TxHashLink/TxHashLink.js]: Did want to put `isTest` into the props.

**Comment by jacogr** [comment]
Happy with the etherscan links now, looks really good and well done. Just don't like the new chain hardcoding that reared its head here.

**Comment by derhuerst** [comment]
> Just don't like the new chain hardcoding that reared its head here.

Thought about giving `txLink` and `addressLink` the possibility to get a chain as second parameter. On the other hand, it's not the responsibility of `3rdparty/etherscan/*` to determine if a certain chain is testnet or not.

**Comment by jacogr** [comment]
100% - the etherscan wrapper doesn't care about the chain, and it shouldn't. 

We already determine the morden/testnet splits elsewhere, which is how we set the isTest props via redux. So instead of just copying that logic newly everywhere, use what we have.

**Pull request merged**